src/chatd.cpp

```diff
@@ -1540,6 +1540,8 @@ void Connection::execCommand(const StaticBuffer& buf)
                 }
                 else
                 {
+                    mChatdClient.mKarereClient->updateAndNotifyLastGreen(msg->userid, msg->ts, false);
```

Comment:
We don't want to execute this line for every `OLDMSG`, it should be only called `if (opcode == OP_NEWMSG))`

Reply:
In any case, probably it's better to call this method once the message is decrypted. Some management messages initially have the `API_USER`/`COMMANDER` as the `userid` and after parsing they change to the user who performed the action (update privileges, change title...) and we would like to update the timestamp in those cases too, if appropriate.